src/screens/Loading.js

```diff
@@ -63,10 +55,33 @@ export default class Loading extends React.PureComponent {
           })
         ]
       });
-      this.props.navigation.dispatch(tocAction);
-      return;
+    } else {
+      tocActions = firstScreenActions;
     }
-    this.props.navigation.dispatch(firstScreenActions);
+    
+    await loadAccounts()
+    this.props.navigation.dispatch(tocActions);
+  }
+
+  async migrateAccount_v1 () {
+    const oldAccounts_v1 = await loadAccounts(1);
+    const accounts = oldAccounts_v1.map(empty).map(a => ({ ...a, v1recov: true }));
+    accounts.forEach(saveAccount);
+  }
+
+  // only ethereum account with chainId and networkType properties
+  async migrateAccount_v2 () {
+    const oldAccounts_v2 = await loadAccounts(2);
+    const accounts = oldAccounts_v2.map(empty).map(a => {
+      let result = a
+      if (a.chainId) {
+        result = { ...a, networkKey: a.chainId, v2recov: true };
```

Comment:
result is already pointing to a, `result = { networkKey: result.chainId, v2recov: true, ...result }` would be more readable, or just keep this but set result to `{}` initially.